### PR TITLE
nightly: fail nightly tests if any cockroach processes exit

### PR DESCRIPTION
**Pull request by danhhz**
This is accomplished by watching the output of `roachprod monitor`.

    --- FAIL: TestSingleDC/kv95 (274.69s)
    	nightly_test.go:157: MONITOR: 3: dead

Release note: None

### CODE REVIEW

File: pkg/nightly/nightly_test.go [hunk 1]
```diff
@@ -122,16 +136,26 @@ func TestSingleDC(t *testing.T) {
...
 				t.Fatal(err)
 			}
+
+			m := monitorWithContext(ctx)
+			m.Worker(func(ctx context.Context) error {
+				// TODO(dan): runCmd internally uses exec.CommandContext, which
+				// is supposed to kill the exec'd process if the context is
+				// canceled. This appears to work generally, but not for
+				// `roachprod ssh`.
+				return runCmd(ctx, l, "roachprod", "ssh", clusterName+":1", "--", testCmd)
```
Comment (benesch) [inline pkg/nightly/nightly_test.go]: I think I fixed this in cockroachdb/roachprod#52.

Comment (danhhz) [reply pkg/nightly/nightly_test.go]: Same problem.

Comment (benesch) [reply pkg/nightly/nightly_test.go]: Womp womp.

**Comment by petermattis** [comment]
:lgtm:

---

Review status: 0 of 1 files reviewed at latest revision, 1 unresolved discussion, all commit checks successful.

---

*[pkg/nightly/nightly_test.go, line 154 at r1](https://reviewable.io:443/reviews/cockroachdb/cockroach/22485#-L4mN-p9e4yFj5-9Nsie-r1-154:-L4mtiV3a7soH4YMLzxu:b-my7jkx) ([raw file](https://github.com/cockroachdb/cockroach/blob/2695453822f49693a54f11c26290c76ef287f649/pkg/nightly/nightly_test.go#L154)):*
<details><summary><i>Previously, benesch (Nikhil Benesch) wrote…</i></summary><blockquote>

Womp womp.
</blockquote></details>

Looks like canceling the context kills the child process with `kill -9`. For most `roachprod` commands this is fine, but `roachprod ssh` is creating a child process. One thought is that perhaps we change `roachprod ssh` to use `syscall.Exec` instead of `exec.Command`. That is, replace the `fork/exec` with a plain `exec`. 

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/22485#-:-L4mvndnpe0LGVEefil7:bnfp4nl)*

**Comment by petermattis** [comment]
Review status: 0 of 1 files reviewed at latest revision, 1 unresolved discussion, all commit checks successful.

---

*[pkg/nightly/nightly_test.go, line 154 at r1](https://reviewable.io:443/reviews/cockroachdb/cockroach/22485#-L4mN-p9e4yFj5-9Nsie-r1-154:-L4mxL9c1GC08Roiuz71:bxhosux) ([raw file](https://github.com/cockroachdb/cockroach/blob/2695453822f49693a54f11c26290c76ef287f649/pkg/nightly/nightly_test.go#L154)):*
<details><summary><i>Previously, petermattis (Peter Mattis) wrote…</i></summary><blockquote>

Looks like canceling the context kills the child process with `kill -9`. For most `roachprod` commands this is fine, but `roachprod ssh` is creating a child process. One thought is that perhaps we change `roachprod ssh` to use `syscall.Exec` instead of `exec.Command`. That is, replace the `fork/exec` with a plain `exec`. 
</blockquote></details>

@danhhz Does this work https://github.com/cockroachdb/roachprod/pull/56?

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/22485)*

**Comment by danhhz** [comment]
Review status: 0 of 1 files reviewed at latest revision, 1 unresolved discussion, some commit checks failed.

---

*[pkg/nightly/nightly_test.go, line 154 at r1](https://reviewable.io:443/reviews/cockroachdb/cockroach/22485#-L4mN-p9e4yFj5-9Nsie-r1-154:-L4pgUnNrMc5HNdkKPRP:bcxu2rg) ([raw file](https://github.com/cockroachdb/cockroach/blob/2695453822f49693a54f11c26290c76ef287f649/pkg/nightly/nightly_test.go#L154)):*
<details><summary><i>Previously, petermattis (Peter Mattis) wrote…</i></summary><blockquote>

@danhhz Does this work https://github.com/cockroachdb/roachprod/pull/56?
</blockquote></details>

I'll try it out today

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/22485)*

**Comment by danhhz** [comment]
Review status: 0 of 1 files reviewed at latest revision, 1 unresolved discussion.

---

*[pkg/nightly/nightly_test.go, line 154 at r1](https://reviewable.io:443/reviews/cockroachdb/cockroach/22485#-L4mN-p9e4yFj5-9Nsie-r1-154:-L4qKrSMHyVIAEGPuNpo:b-93o0ad) ([raw file](https://github.com/cockroachdb/cockroach/blob/2695453822f49693a54f11c26290c76ef287f649/pkg/nightly/nightly_test.go#L154)):*
<details><summary><i>Previously, danhhz (Daniel Harrison) wrote…</i></summary><blockquote>

I'll try it out today
</blockquote></details>

Success! Removing TODOs

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/22485)*

**Comment by danhhz** [comment]
TFTRs!

**Pull request merged**